From b26025c1b6a826e3c7873bd6d3b142391cb9ab8b Mon Sep 17 00:00:00 2001
From: Torsten Grote <t@grobox.de>
Date: Thu, 18 Jun 2020 13:15:16 -0300
Subject: [PATCH] Show privacy warning on in-call screen

Includes:
Author: Chirayu Desai <chirayudesai1@gmail.com>
Date:   Tue Jul 28 02:23:37 2020 +0530

    Move the incall warning back to the center

    * Instead of left/started aligned.

Change-Id: Ic9b18c110481df7042ca1daa36182e0999fe948f
---
 .../dialer/theme/common/res/values/colors.xml |  3 +-
 .../res/layout/fragment_incoming_call.xml     |  7 +++-
 .../contactgrid/res/layout/incall_header.xml  | 39 +++++++++++++++++++
 .../impl/res/layout/frag_incall_voice.xml     |  7 ++++
 .../android/incallui/res/values/strings.xml   |  3 ++
 .../incallui/theme/res/values/styles.xml      |  4 +-
 6 files changed, 59 insertions(+), 4 deletions(-)
 create mode 100644 java/com/android/incallui/contactgrid/res/layout/incall_header.xml

diff --git a/java/com/android/dialer/theme/common/res/values/colors.xml b/java/com/android/dialer/theme/common/res/values/colors.xml
index 1831862b1..58316d0de 100644
--- a/java/com/android/dialer/theme/common/res/values/colors.xml
+++ b/java/com/android/dialer/theme/common/res/values/colors.xml
@@ -21,6 +21,7 @@
   <color name="dialer_divider_line_color">#D8D8D8</color>
   <color name="dialer_link_color">#2A56C6</color>
   <color name="dialer_snackbar_action_text_color">#4285F4</color>
+  <color name="incall_header_background">#f3b514</color>
 
   <!-- Colors for the notification actions -->
   <color name="notification_action_accept">#097138</color>
diff --git a/java/com/android/incallui/answer/impl/res/layout/fragment_incoming_call.xml b/java/com/android/incallui/answer/impl/res/layout/fragment_incoming_call.xml
index 1672bf984..63de75999 100644
--- a/java/com/android/incallui/answer/impl/res/layout/fragment_incoming_call.xml
+++ b/java/com/android/incallui/answer/impl/res/layout/fragment_incoming_call.xml
@@ -78,18 +78,23 @@
         android:id="@+id/incall_contact_grid"
         android:layout_width="match_parent"
         android:layout_height="match_parent"
-        android:layout_marginTop="24dp"
         android:clipChildren="false"
         android:clipToPadding="false"
         android:focusable="true"
         android:gravity="top|center_horizontal"
         android:orientation="vertical">
 
+      <include
+          layout="@layout/incall_header"
+          android:layout_width="match_parent"
+          android:layout_height="wrap_content"/>
+
       <include
           android:id="@id/contactgrid_top_row"
           layout="@layout/incall_contactgrid_top_row"
           android:layout_width="wrap_content"
           android:layout_height="wrap_content"
+          android:layout_marginTop="24dp"
           android:layout_marginBottom="8dp"
           android:layout_marginStart="24dp"
           android:layout_marginEnd="24dp"/>
diff --git a/java/com/android/incallui/contactgrid/res/layout/incall_header.xml b/java/com/android/incallui/contactgrid/res/layout/incall_header.xml
new file mode 100644
--- /dev/null
+++ b/java/com/android/incallui/contactgrid/res/layout/incall_header.xml
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (C) 2020 The Calyx Institute
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~      http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License
+  -->
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:background="@color/incall_header_background"
+    android:gravity="center"
+    android:orientation="vertical"
+    android:padding="8dp"
+    android:fitsSystemWindows="true">
+
+  <TextView
+      android:id="@+id/incall_privacy_title"
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:textColor="?android:attr/textColorPrimaryInverse"
+      android:drawableStart="@drawable/ic_baseline_warning"
+      android:drawablePadding="8dp"
+      android:drawableTint="?android:attr/textColorPrimaryInverse"
+      android:gravity="center"
+      android:padding="4dp"
+      android:text="@string/incall_screen_privacy_data_collection"/>
+
+</LinearLayout>
diff --git a/java/com/android/incallui/incall/impl/res/layout/frag_incall_voice.xml b/java/com/android/incallui/incall/impl/res/layout/frag_incall_voice.xml
--- a/java/com/android/incallui/incall/impl/res/layout/frag_incall_voice.xml
+++ b/java/com/android/incallui/incall/impl/res/layout/frag_incall_voice.xml
@@ -29,11 +29,18 @@
       android:clipToPadding="false"
       android:fitsSystemWindows="true">
 
+    <include
+        layout="@layout/incall_header"
+        android:id="@+id/incall_header"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"/>
+
     <LinearLayout
         android:id="@id/incall_contact_grid"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         android:layout_marginTop="12dp"
+        android:layout_below="@+id/incall_header"
         android:gravity="center_horizontal"
         android:orientation="vertical">
 
diff --git a/java/com/android/incallui/res/values/strings.xml b/java/com/android/incallui/res/values/strings.xml
--- a/java/com/android/incallui/res/values/strings.xml
+++ b/java/com/android/incallui/res/values/strings.xml
@@ -212,4 +212,7 @@
   <!-- Text for button to accept RTT request. [CHAR LIMIT=20] -->
   <string name="rtt_button_accept_request">Join RTT</string>
 
+  <!-- Text for call not private banner. -->
+  <string name="incall_screen_privacy_data_collection">The location and audio of this call are not private.</string>
+
 </resources>
diff --git a/java/com/android/incallui/theme/res/values/styles.xml b/java/com/android/incallui/theme/res/values/styles.xml
--- a/java/com/android/incallui/theme/res/values/styles.xml
+++ b/java/com/android/incallui/theme/res/values/styles.xml
@@ -21,7 +21,7 @@
        Theme.Black.NoTitleBar directly, since we want any popups or dialogs from the
        InCallActivity to have the correct Material style. -->
   <style name="Theme.InCallScreen.Light" parent="@style/Dialer.ThemeBase.NoActionBar">
-    <item name="android:statusBarColor">@android:color/transparent</item>
+    <item name="android:statusBarColor">@color/incall_header_background</item>
     <item name="android:navigationBarColor">@android:color/transparent</item>
     <item name="android:windowDrawsSystemBarBackgrounds">true</item>
 
@@ -36,7 +36,7 @@
   <style name="Theme.InCallScreen" parent="@style/Dialer.Dark.ThemeBase.NoActionBar">
     <item name="android:textColorPrimary">@android:color/white</item>
     <item name="android:textColorSecondary">#DDFFFFFF</item>
-    <item name="android:statusBarColor">@android:color/transparent</item>
+    <item name="android:statusBarColor">@color/incall_header_background</item>
     <item name="android:navigationBarColor">@android:color/transparent</item>
     <item name="android:windowDrawsSystemBarBackgrounds">true</item>
     <item name="android:colorPrimaryDark">@color/dialer_theme_color_dark</item>
-- 
2.31.1

